packages/mobile/src/geth/geth.ts

```diff
@@ -123,71 +126,71 @@ export async function initGeth(sync: boolean = true): Promise<typeof gethInstanc
+      if (!(await ensureGenesisBlockWritten())) {
+        throw FailedToFetchGenesisBlockError
+      }
+      if (!(await ensureStaticNodesInitialized())) {
+        throw FailedToFetchStaticNodesError
+      }
     }
 
-    let geth: typeof RNGeth
     ValoraAnalytics.track(GethEvents.create_geth_start)
     try {
-      geth = await createNewGeth(sync)
+      gethInstance = await createNewGeth(shouldStartNode)
     } catch (error) {
       ValoraAnalytics.track(GethEvents.create_geth_error, { error: error.message })
       throw error
     }
     ValoraAnalytics.track(GethEvents.create_geth_finish)
 
-    if (!sync) {
-      // chain data must be deleted to prevent geth from syncing with data saver on
-      // TODO: consider only deleting the geth p2p nodes database
-      // TODO: save chain data s.t. when data saver goes off syncing resumes from data
-      await deleteChainData()
-    }
-
-    try {
-      ValoraAnalytics.track(GethEvents.start_geth_start)
-      await geth.start()
-      ValoraAnalytics.track(GethEvents.start_geth_finish)
-      gethInstance = geth
-      if (sync) {
-        geth.subscribeNewHead()
-      }
-    } catch (e) {
-      const errorType = getGethErrorType(e)
-      if (errorType === ErrorType.GethAlreadyRunning) {
-        Logger.error('Geth@init/startInstance', 'Geth start reported geth already running')
-        throw new Error('Geth already running, need to restart app')
-      } else if (errorType === ErrorType.CorruptChainData) {
-        Logger.warn('Geth@init/startInstance', 'Geth start reported chain data error')
-        await attemptGethCorruptionFix(geth, sync)
-      } else {
-        Logger.error('Geth@init/startInstance', 'Unexpected error starting geth', e)
-        throw e
+    if (shouldStartNode) {
+      try {
+        ValoraAnalytics.track(GethEvents.start_geth_start)
+        await gethInstance.start()
+        ValoraAnalytics.track(GethEvents.start_geth_finish)
+        await gethInstance.subscribeNewHead()
+      } catch (e) {
+        const errorType = getGethErrorType(e)
+        if (errorType === ErrorType.GethAlreadyRunning) {
+          Logger.error('Geth@init/startInstance', 'Geth start reported geth already running')
+          throw new Error('Geth already running, need to restart app')
+        } else if (errorType === ErrorType.CorruptChainData) {
+          Logger.warn('Geth@init/startInstance', 'Geth start reported chain data error')
+          await attemptGethCorruptionFix(gethInstance, shouldStartNode)
+        } else {
+          Logger.error('Geth@init/startInstance', 'Unexpected error starting geth', e)
+          throw e
+        }
       }
     }
-
     return gethInstance
   } finally {
     gethLock = false
   }
 }
 
+// TODO: This needs a bigger change to the design
+// I'm exposing this now in order to build a proof
+// of concept but I need input from the wallet team
+// regarding what's the best way to expose the instance.
+export function getGethInstance() {
```

Comment:
Can the signer work without the geth client running at all in Data Saver mode? If so, I think this would be cleaner if all geth-related logic were skipped / not initialized when in Data Saver mode

Reply:
Resolving based on our chat